apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js

```diff
@@ -91,96 +95,91 @@ const prepareAndSendNotification = async (context, receipt, resident) => {
  * Reads by portions all data required to send notifications to residents on available billing receipts and sends notifications
  * @param receiptsWhere
  * @param onLastDtChange
+ * @param taskId
  * @returns {Promise<void>}
  */
-const sendBillingReceiptsAddedNotificationsForPeriod = async (receiptsWhere, onLastDtChange) => {
+const sendBillingReceiptsAddedNotificationsForPeriod = async (receiptsWhere, onLastDtChange, taskId) => {
     const { keystone: context } = await getSchemaCtx('Resident')
-    const receiptsCount = await BillingReceipt.count(context, receiptsWhere)
-    let skip = 0, successCount = 0
+    let successCount = 0
     const notifiedUsers = new Set()
 
-    logger.info({ msg: 'sending billing receipts', receiptsCount, data: receiptsWhere })
-
-    // Exit if no receipts found to proceed
-    if (!receiptsCount) return
-
     let lastReceipt
 
-    while (skip < receiptsCount) {
-        const receipts = await BillingReceipt.getAll(context, receiptsWhere, { sortBy: ['createdAt_ASC'], first: CHUNK_SIZE, skip })
-
-        if (isEmpty(receipts)) break
-
-        skip += receipts.length
+    await processListByChunks({
+        context,
+        chunkSize: CHUNK_SIZE,
+        list: BillingReceipt,
+        where: receiptsWhere,
+        chunkProcessor: async (/** @type {BillingReceipt[]} */ receipts) => {
+            const payableReceipts = receipts.filter(receipt => receipt.isPayable)
+            let organisationIds = []
+            const accountsNumbers = []
+            for (const receipt of payableReceipts) {
+                organisationIds.push(get(receipt, 'context.organization.id'))
+                accountsNumbers.push(get(receipt, 'account.number'))
+            }
+            organisationIds = uniq(organisationIds)
 
-        let organisationIds = []
-        const accountsNumbers = []
-        for (const receipt of receipts) {
-            organisationIds.push(get(receipt, 'context.organization.id'))
-            accountsNumbers.push(get(receipt, 'account.number'))
-        }
-        organisationIds = uniq(organisationIds)
+            const serviceConsumerWhere = {
+                organization: { id_in: organisationIds },
+                accountNumber_in: accountsNumbers,
```

Comment:
what will happen if `accountsNumbers` array has one or many "undefined" values? won't it return any accountNumber?
ex. ['123', undefined, '323'] -> because of "undefined"  it'll return serviceConsumers with weird accountNumbers
i might be wrong tho, just checking